### ISSUE TITLE
Unified v11 has been released, can remark be updated to support it?

### ISSUE THREAD

**Issue by favna** [remarkjs/remark#1219]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

Unified 11.x was released last week (https://github.com/unifiedjs/unified/releases/tag/11.0.0). Installing it right now gives TypeScript compilation errors:

![ishare-1692800039](https://github.com/remarkjs/remark/assets/4019718/c97a8376-c0db-41fe-bf68-dccb0521199d)

### Solution

Remark libraries should be updated to work with and use Unified v11

### Alternatives

N.A. Technically there is "keep using Unified 10" but it's ultimately just a matter of time for that to become an issue.

**Comment by wooorm** [comment]
Yes. See https://github.com/unifiedjs/unified/issues/228, https://github.com/unifiedjs/unified/issues/227!
